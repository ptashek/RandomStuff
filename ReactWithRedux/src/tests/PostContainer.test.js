import React from 'react';
import PostContainer from '../components/PostContainer.react';
import renderer from 'react-test-renderer';

import { TestPost, emptyFunction } from './TestData';

it('renders ok if inactive', () => {
  const component = renderer.create(
    <PostContainer
      onClick={emptyFunction}
      active={false}
      post={TestPost}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children.length).toBe(2);

  tree.children[0].props.onClick();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ok if active', () => {
  const component = renderer.create(
    <PostContainer
      onClick={emptyFunction}
      active={true} 
      post={TestPost}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children.length).toBe(2);

  tree.children[0].props.onClick();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
