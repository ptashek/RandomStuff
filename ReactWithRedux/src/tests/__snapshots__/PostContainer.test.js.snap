exports[`test renders ok if active 1`] = `
<div>
  <div
    className="Post-Title Post-Title-Active"
    onClick={[Function]}>
    title
  </div>
  <div
    className="Post-Body">
    body
  </div>
</div>
`;

exports[`test renders ok if active 2`] = `
<div>
  <div
    className="Post-Title Post-Title-Active"
    onClick={[Function]}>
    title
  </div>
  <div
    className="Post-Body">
    body
  </div>
</div>
`;

exports[`test renders ok if inactive 1`] = `
<div>
  <div
    className="Post-Title"
    onClick={[Function]}>
    title
  </div>
  <div
    className="Post-Body">
    body
  </div>
</div>
`;

exports[`test renders ok if inactive 2`] = `
<div>
  <div
    className="Post-Title"
    onClick={[Function]}>
    title
  </div>
  <div
    className="Post-Body">
    body
  </div>
</div>
`;
